components/engine_traits/src/flush.rs

```diff
@@ -54,6 +54,40 @@ struct FlushProgress {
     last_flushed: [u64; DATA_CFS_LEN],
 }
 
+/// A share state between raftstore and underlying engine.
+///
+/// raftstore will update state changes and corresponding sst apply index, when
+/// apply ingest sst request, it should ensure the sst can be deleted
+/// if the flushed index greater than it .
+#[derive(Debug, Clone)]
+pub struct SstApplyState {
+    sst_map: Arc<RwLock<HashMap<Vec<u8>, u64>>>,
+}
+
+impl Default for SstApplyState {
+    fn default() -> Self {
+        Self {
+            sst_map: Arc::new(RwLock::new(HashMap::new())),
+        }
+    }
+}
+
+impl SstApplyState {
+    #[inline]
+    pub fn registe_ssts(&self, uuids: Vec<Vec<u8>>, sst_applied_index: u64) {
+        let mut map = self.sst_map.write().unwrap();
+        for uuid in uuids {
+            map.insert(uuid, sst_applied_index);
+        }
+    }
+
+    /// Query the sst applied index.
+    #[inline]
+    pub fn sst_applied_index(&self, uuid: &Vec<u8>) -> Option<u64> {
+        self.sst_map.read().unwrap().get(uuid).copied()
```

Comment:
Do we need to GC `sst_map`?

Reply:
yes, I will remove the uuid which needs to delete?